alicloud/resource_alicloud_drds_instance.go

```diff
@@ -0,0 +1,174 @@
+package alicloud
+
+import (
+	"fmt"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/drds"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"strconv"
+	"time"
+)
+
+func resourceAliCloudDRDSInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliCloudDRDSInstanceCreate,
+		Read:   resourceAliCloudDRDSInstanceRead,
+		Update: resourceAliCloudDRDSInstanceUpdate,
+		Delete: resourceAliCloudDRDSInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateStringLengthInRange(1, 129),
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validateAllowedStringValue([]string{string(PrivateType), string(PrivateType_),
+					string(PublicType), string(PublicType_)}),
+			},
+			"zone_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"specification": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"pay_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue([]string{string(DRDSInstancePostPayType)}),
+			},
+			"vpc_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"vswitch_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"instance_series": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			//`quantity` is similar to `count`, still not sure how to combine these two parameters.
+			"quantity": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAliCloudDRDSInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient).drdsconn
+
+	req := drds.CreateCreateDrdsInstanceRequest()
+	req.Description = d.Get("description").(string)
+	req.Type = d.Get("type").(string)
+	req.ZoneId = d.Get("zone_id").(string)
+	req.Specification = d.Get("specification").(string)
+	req.PayType = d.Get("pay_type").(string)
+	req.VpcId = d.Get("vpc_id").(string)
+	req.VswitchId = d.Get("vswitch_id").(string)
+	req.InstanceSeries = d.Get("instance_series").(string)
+	req.Quantity = strconv.Itoa(d.Get("quantity").(int))
+
+	response, err := client.CreateDrdsInstance(req)
+	idList := response.Data.DrdsInstanceIdList.DrdsInstanceId
```

Comment:
Please give priority to judge err. If response is nil, line 87 will be a empty pointer error.

Reply:
Can `response ` be `nil`?
https://help.aliyun.com/document_detail/51124.html?spm=a2c4g.11174283.6.733.PCDcpE